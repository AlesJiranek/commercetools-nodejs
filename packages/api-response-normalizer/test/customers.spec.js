import normalizeCustomers from '../src/customers'
import customersResponse from './fixtures/customers.json'

describe('when the response is from `/customers`', () => {
  it('should normalize the results', () => {
    expect(normalizeCustomers(customersResponse)).toMatchSnapshot()
  })
  it('should return a normalized shape', () => {
    expect(normalizeCustomers(customersResponse)).toEqual(
      expect.objectContaining({
        entities: expect.objectContaining({
          customerGroups: expect.any(Object),
          customers: expect.any(Object),
        }),
        result: expect.objectContaining({
          results: expect.any(Array),
        }),
      })
    )
  })
})
