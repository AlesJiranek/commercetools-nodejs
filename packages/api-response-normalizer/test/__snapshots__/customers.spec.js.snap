// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`when the response is from \`/customers\` should normalize the results 1`] = `
Object {
  "entities": Object {
    "customerGroups": Object {
      "0060effa-ee62-4adb-a455-5893dfb5c617": Object {
        "createdAt": "2016-08-29T16:25:09.690Z",
        "id": "0060effa-ee62-4adb-a455-5893dfb5c617",
        "lastModifiedAt": "2016-08-29T16:25:09.690Z",
        "name": "B2C",
        "version": 1,
      },
    },
    "customers": Object {
      "0db6f3c1-c82c-4398-942a-4eaaf2b99907": Object {
        "addresses": Array [],
        "billingAddressIds": Array [],
        "companyName": "sgsdfgd",
        "createdAt": "2016-09-09T15:49:42.216Z",
        "email": "foo+1@bar.com",
        "externalId": "f123",
        "firstName": "fsdgdfg",
        "id": "0db6f3c1-c82c-4398-942a-4eaaf2b99907",
        "isEmailVerified": false,
        "lastMessageSequenceNumber": 1,
        "lastModifiedAt": "2016-11-23T19:50:42.794Z",
        "password": "foo",
        "shippingAddressIds": Array [],
        "version": 2,
      },
      "1c310cc9-25d9-4b38-bf09-75afa6d75295": Object {
        "addresses": Array [],
        "billingAddressIds": Array [],
        "createdAt": "2016-09-09T15:32:33.593Z",
        "email": "john@doe.com",
        "firstName": "John",
        "id": "1c310cc9-25d9-4b38-bf09-75afa6d75295",
        "isEmailVerified": false,
        "lastMessageSequenceNumber": 1,
        "lastModifiedAt": "2016-09-09T15:32:33.593Z",
        "lastName": "Doe",
        "password": "foo",
        "shippingAddressIds": Array [],
        "title": "Mr",
        "version": 1,
      },
      "599f176f-d6e1-4e44-b5b0-edf668d1c57e": Object {
        "addresses": Array [],
        "billingAddressIds": Array [],
        "createdAt": "2016-09-09T15:46:11.472Z",
        "email": "foo@bar.com",
        "id": "599f176f-d6e1-4e44-b5b0-edf668d1c57e",
        "isEmailVerified": false,
        "lastMessageSequenceNumber": 1,
        "lastModifiedAt": "2016-09-09T15:46:11.472Z",
        "password": "foo",
        "shippingAddressIds": Array [],
        "version": 1,
      },
      "9199f748-9d75-448a-8d06-d60a6c126ce6": Object {
        "addresses": Array [
          Object {
            "apartment": "ccc",
            "building": "c",
            "city": "ccccc",
            "company": "sfdf",
            "country": "AD",
            "firstName": "svb",
            "id": "fD2hQNpd",
            "lastName": "sfsf",
            "pOBox": "cccc",
            "postalCode": "ccc",
            "salutation": "ds",
            "streetName": "ccccc",
            "streetNumber": "2",
            "title": "fsdfs",
          },
        ],
        "billingAddressIds": Array [],
        "companyName": "example",
        "createdAt": "2016-09-09T16:08:11.846Z",
        "customerGroup": Object {
          "id": "0060effa-ee62-4adb-a455-5893dfb5c617",
          "obj": "0060effa-ee62-4adb-a455-5893dfb5c617",
          "typeId": "customer-group",
        },
        "customerNumber": "123456",
        "email": "alicia.williams31@example.com",
        "externalId": "aaa111",
        "firstName": "Alicia",
        "id": "9199f748-9d75-448a-8d06-d60a6c126ce6",
        "isEmailVerified": false,
        "lastMessageSequenceNumber": 1,
        "lastModifiedAt": "2016-11-17T09:45:24.962Z",
        "lastName": "Williams",
        "password": "foo",
        "shippingAddressIds": Array [],
        "title": "Mrs",
        "version": 2,
      },
      "9e0de50f-90d7-4f6f-bded-a0731a8caa3e": Object {
        "addresses": Array [],
        "billingAddressIds": Array [],
        "createdAt": "2016-09-09T15:42:47.902Z",
        "email": "aaa@bbb.com",
        "firstName": "aaa",
        "id": "9e0de50f-90d7-4f6f-bded-a0731a8caa3e",
        "isEmailVerified": false,
        "lastMessageSequenceNumber": 1,
        "lastModifiedAt": "2016-09-09T15:42:47.902Z",
        "lastName": "bbb",
        "password": "foo",
        "shippingAddressIds": Array [],
        "title": "Mr",
        "version": 1,
      },
    },
  },
  "result": Object {
    "count": 5,
    "limit": 5,
    "offset": 0,
    "results": Array [
      "9199f748-9d75-448a-8d06-d60a6c126ce6",
      "0db6f3c1-c82c-4398-942a-4eaaf2b99907",
      "1c310cc9-25d9-4b38-bf09-75afa6d75295",
      "9e0de50f-90d7-4f6f-bded-a0731a8caa3e",
      "599f176f-d6e1-4e44-b5b0-edf668d1c57e",
    ],
    "total": 105,
  },
}
`;
